test/functional/test_runner.py

```diff
@@ -471,6 +474,13 @@ def main():
 
     logging.debug("Temporary test directory at %s" % tmpdir)
 
+    results_filepath = None
+    if args.resultsfile:
+        results_filepath = pathlib.Path(args.resultsfile)
```

Comment:
The write function below `write_results` intentionally creates a csv file, should we add a check here for the `csv` file extension?

Reply:
As soon as the `functional_test_results.csv` file was generated in my system, I thought of adding it in `gitignore`. Since this feature is conditional based on the param, we can't add this hardcoded name in gitgnore but how about adding `*.csv` in gitgnore?

Reply:
Good thought.  Since the approach is to allow the user to specify the filename/extension as they see fit (rather than force a particular extension), this seems ok as-is.  If others feel that forcing the extension `.csv` would be more ideal, then will keep this in mind.